common/src/main/java/com/distkv/dst/common/id/NodeId.java

```diff
@@ -0,0 +1,78 @@
+package com.distkv.dst.common.id;
+
+import java.io.Serializable;
+
+/**
+ * The class that represents the Id of a node.
+ */
+public class NodeId implements Serializable {
+
+  private static final long serialVersionUID = -2216426913363839949L;
+
+  /**
+   * Whether this node is a master.
+   *
+   * True if this node is master node, false if this node is a slave node.
+   */
+  private boolean isMaster;
+
+  /**
+   * The Id of the partition that this node belongs to.
+   */
+  private PartitionId partitionId;
+
+  /**
+   * The index of this node.
+   */
+  private int index;
+
+  public boolean isMaster() {
+    return isMaster;
+  }
+
+  public PartitionId getPartitionId() {
+    return partitionId;
+  }
+
+  public int getIndex() {
+    return index;
+  }
+
+  private NodeId(int index, PartitionId partitionId, boolean isMaster) {
+    this.index = index;
+    this.partitionId = partitionId;
+    this.isMaster = isMaster;
+  }
+
+  public static NodeId from(int index, PartitionId partitionId, boolean isMaster) {
+    return new NodeId(index, partitionId, isMaster);
+  }
+
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == null) {
+      return false;
+    }
+    if (!this.getClass().equals(obj.getClass())) {
```

Comment:
I think it will be clearer using instanceof.

Reply:
What do you think?

Reply:
Thanks.

It could not be `instanceof` here because `instanceof` couldn't distinguish the base type.

For example, if I add a class `MyNodeId` that extends from `NodeId`:
```java
class MyNodeId extends NodeId {
// ...
}
```
Considering this case:
```java
NodeId a = new NodeId();
MyNodeId b = new MyNodeId();
boolean ok = a.equals(b);
```
The result of `ok` is true which is not as expected.

Reply:
Thanks, i think you are right!